packages/opencensus-web-core/src/common/url-util.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Interface to represent the components of a parsed URL.
+ * The field comments below reference this sample URL:
+ *    https://example.com:3000/pathname/?search=test#hash
+ */
+export interface ParsedUrl {
+  /** Protocol of the URL. E.g. https */
+  protocol: string;
+  /** Host name only (no port). E.g. example.com */
+  hostname: string;
+  /** Port number if explicitly specified (else empty string). E.g. 3000 */
+  port: string;
```

Comment:
```number``` instead of ```string```?

Reply:
Well, because this is in a browser context where I want to keep code size small, I'm implementing this by generating an `<a>` tag, which uses `string` for the port (and leaves it as an empty string if it is not explicitly specified). I added a comment to explain that.